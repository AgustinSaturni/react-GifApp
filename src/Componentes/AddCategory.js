import React,{useState} from 'react'
import PropTypes from 'prop-types';


export const AddCategory = ({setCategorias}) => { //recibo funcion que mandÃ©
    const [valorInicial, setState] = useState(''); //defino el estado del componente

    const handleChange = (event)=>{      //Este me actualiza mi estado con cada char que escribo en el box.

        setState(event.target.value)

    }

    const handleSubmit=(event)=>{       //este usa la funcion que viene como argumento y agrega al estado del otro componente un elemento mas.
        event.preventDefault();
        if(valorInicial.trim().length >2){
            setCategorias(categorias => [valorInicial,...categorias]);
            setState('');
        }
      
    }

    return (            //este onSubmit actua cuando aprieto enter.
        <form onSubmit={handleSubmit}> 
            <p>{valorInicial}</p>
            <input type="text" value={valorInicial} onChange={handleChange} />
        </form>
    )
}


AddCategory.propTypes={
    setCategorias: PropTypes.func.isRequired

}