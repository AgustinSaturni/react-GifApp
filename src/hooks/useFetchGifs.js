import { useEffect, useState } from "react"
import {getGifs} from '../helpers/getGifs'


export const useFetchGifs = (categoria) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    })
    
    useEffect(() => {   //Esto se dispara solo cuando cambia la categoria que esta entre corchetes.
       getGifs(categoria).then(img => setState({
           data: img,
           loading: false
       }))
    }, [categoria]);

    return state;

}
