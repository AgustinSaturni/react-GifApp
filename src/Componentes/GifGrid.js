
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';


export const GifGrid = ({categoria}) => {
 
    const {data:images, loading} =useFetchGifs(categoria);


    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{categoria}</h3>
            {loading? <p className="animate__animated animate__flash">Loading</p> : null} 
            <div className="card-grid">
              {
                  images.map(image =>  (<GifGridItem key={image.id} {...image} />) )
              }
           </div>
        </>
    )
}

GifGrid.propTypes={
    categoria: PropTypes.string.isRequired
}