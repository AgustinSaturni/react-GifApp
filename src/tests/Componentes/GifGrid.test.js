import { GifGrid } from "../../Componentes/GifGrid";
import {shallow} from 'enzyme'
import React from 'react'
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs') //finge llamadas al archivo.

describe('Testing en el componente',()=>{
    
    test('Match con el snapshot',()=>{

            useFetchGifs.mockReturnValue({
                data: [],
                loading: true           //valores que el useFetch regresa por defecto
            });
            
            const wrapper = shallow(<GifGrid categoria={"dbz"} />)  //el shallow renderiza un componente como el render.
            expect(wrapper).toMatchSnapshot(); 
    
    })

    test('debe de mostrar Items cuando se cargan imagenes con useFetchGifs',()=>{
        
       const gifs = [{
           id:'ABC',
           url: 'https://localhost/3030',
           title: 'cualquier cosa'
       }]

     useFetchGifs.mockReturnValue({
                data: gifs,
                loading: false           //valores que el useFetch regresa por defecto
            });
            
            const wrapper = shallow(<GifGrid categoria={"dbz"} />)
            expect(wrapper).toMatchSnapshot();
            expect(wrapper.find('p').exists()).toBe(false);
            expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
})

})