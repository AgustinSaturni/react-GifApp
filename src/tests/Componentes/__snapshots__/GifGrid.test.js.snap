// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing en el componente Match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    dbz
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing en el componente debe de mostrar Items cuando se cargan imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    dbz
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://localhost/3030"
    />
  </div>
</Fragment>
`;
